library/std/src/path.rs

```diff
@@ -1411,17 +1411,28 @@ impl PathBuf {
         self.push(file_name);
     }
 
-    /// Updates [`self.extension`] to `extension`.
+    /// Updates [`self.extension`] to `Some(extension)` or to `None` if
+    /// `extension` is empty.
     ///
     /// Returns `false` and does nothing if [`self.file_name`] is [`None`],
     /// returns `true` and updates the extension otherwise.
     ///
     /// If [`self.extension`] is [`None`], the extension is added; otherwise
     /// it is replaced.
     ///
+    /// If `extension` is the empty string, [`self.extension`] will be [`None`]
+    /// afterwards, not `Some("")`.
+    ///
     /// [`self.file_name`]: Path::file_name
     /// [`self.extension`]: Path::extension
     ///
+    /// # Caveats
+    ///
+    /// Note that there are some exceptions when the parameter contains dots
+    /// (then the new [`self.extension`] will only be the last part), or if the
+    /// parameter is the empty string and the file stem also contains internal
+    /// dots.
```

Comment:
This sentence is a bit vague. How about something like this?

```suggestion
    /// The new `extension` may contain dots and will be used in its entirety,
    /// but only the part after the final dot will be reflected in [`self.extension()`][Self::extension].
    /// See the examples below.
```

Reply:
Thanks, added. Also added the part about empty parameter and file stem with internal dots which was missing from your wording.